api/types/types.proto

```diff
@@ -2635,6 +2638,20 @@ message TeamMapping {
     repeated string KubeUsers = 5 [ (gogoproto.jsontag) = "kubernetes_users,omitempty" ];
 }
 
+// TeamRolesMapping represents a single team membership mapping.
+message TeamRolesMapping {
+    // Organization is a Github organization a user belongs to.
+    string Organization = 1 [ (gogoproto.jsontag) = "organization" ];
+    // Team is a team within the organization a user belongs to.
+    string Team = 2 [ (gogoproto.jsontag) = "team" ];
+    // Roles is a list of allowed logins for this org/team.
+    repeated string Roles = 3 [ (gogoproto.jsontag) = "roles,omitempty" ];
+    // KubeGroups is a list of allowed kubernetes groups for this org/team.
+    repeated string KubeGroups = 4 [ (gogoproto.jsontag) = "kubernetes_groups,omitempty" ];
+    // KubeUsers is a list of allowed kubernetes users to impersonate for this org/team.
+    repeated string KubeUsers = 5 [ (gogoproto.jsontag) = "kubernetes_users,omitempty" ];
```

Comment:
Do we still need these fields? I believe they're covered by `Roles`, so presumably we can get rid of them along with `Logins`

https://github.com/gravitational/teleport/blob/5696e0b8e7a4d9d7fa5c42d50f6d7672704db707/api/types/types.proto#L1648-L1654

Reply:
Hmm, that does seem like a good idea if we're going to make this change anyway. What do you think @benarent ?

Reply:
I don't think we can get rid of these right now. Doing so is quite the breaking change, we should deprecate them first I think.